samples/SortBenchmark/SortBenchmark.fs

```diff
@@ -390,98 +266,310 @@ type RemoteFunc( filePartNum:int, records:int64, _dim:int , partNumS1:int, partN
+                            Buffer.BlockCopy(tbuf,0,ttbuf,0,tbuf.Length)
+                            memBuf.Write(tbuf,0,toRead)
+                            totalReadLen := !totalReadLen + (int64) toRead
+
+                            counter := !counter + 1
+                            if (!counter % 100 = 0) then
+                                Logger.LogF( LogLevel.MildVerbose, ( fun _ -> sprintf "Read %d bytes from file" !totalReadLen) )
+                            x.diskHelper.ReportReadBytes((int64)toRead)
+                            yield memBuf
                 }
-            Logger.LogF( LogLevel.WildVerbose, (fun _ -> sprintf "UTC %s, all data from file %s has been read" (UtcNowToString()) "adf"))
             ret
 
+    member val repartitionThread = ref 0
 
-    member x.WritePreSortFuncFakeNativeRepSeq ( data ) = 
-        data |> Seq.iter(fun (i,buf,len) -> 
-        
 
-                        let oBuf:byte[] = x.GetRepartitionMemBuf()
-                        let r = Interop.NativeBin(buf,len,x.dim,partNumS1,stageTwoPartionBoundary,oBuf)
-        
-                        RemoteFunc.partiSharedMem.Enqueue(buf)
-                        RemoteFunc.partiSharedMem.Enqueue(oBuf)
-                        
-                        
-                        )
-        //data |> Seq.iter(fun (i,d) -> ())
-        ()
+    member internal x.RepartitionMemStream (stage:RepartitionStage) (buffer:MemoryStreamB) = 
+        if buffer.Length > 0L then
+            let retseq = seq {
 
+                        let partionBoundary =
+                            if stage = RepartitionStage.StageOne then
+                                stageOnePartionBoundary
+                            else 
+                                stageTwoPartionBoundary    
 
+                        let nump = 
+                            if stage = RepartitionStage.StageOne then
+                                partNumS1
+                            else 
+                                partNumS2
 
-    member x.WritePreSortFuncFakeSeq ( data ) = 
-        let tbuf = Array.zeroCreate<byte> x.dim
-        data |> Seq.iter(fun (i,buf:byte[],len) -> 
+                        let partstream = Array.init<MemoryStreamB> nump (fun i -> 
+                                                                            null
+                                                                            )
 
-                        let nump = partNumS1
-                        let posi = Array.zeroCreate<int> nump
+                        let t1 = DateTime.UtcNow
+                        let bHasBuf = ref true
+                        let sr = new StreamReader<byte>(buffer,0L)
+
+                        let bRemainRecord = ref false
+                        let remainBuf = ref Unchecked.defaultof<byte[]>
+                        let remainpos = ref 0
+                        let remainLen = ref 0
+
+                        while !bHasBuf do
```

Comment:
Seems you really care about perf. I think you'll see some benefit from "let mutable", not to mention less noisy syntax.